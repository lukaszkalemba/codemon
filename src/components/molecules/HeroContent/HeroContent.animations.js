const getHeading = size => {
  return {
    variants: {
      initial: { opacity: 0, x: size.width > 1150 ? 80 : -80 },
      animate: { opacity: 1, x: 0, transition: { duration: 0.75 } },
    },
  };
};

const getParagraph = size => {
  return {
    variants: {
      initial: { opacity: 0, x: size.width > 1150 ? -80 : 80 },
      animate: { opacity: 0.5, x: 0, transition: { duration: 0.75 } },
    },
  };
};

const getButtons = size => {
  return {
    variants: {
      initial: { opacity: 0, x: size.width > 1150 ? -80 : 80 },
      animate: {
        opacity: 1,
        x: 0,
        transition: {
          duration: 0.75,
          delay: size.width > 1150 ? 1 : 2,
        },
      },
    },
  };
};

export default {
  getHeading,
  getParagraph,
  getButtons,
};
