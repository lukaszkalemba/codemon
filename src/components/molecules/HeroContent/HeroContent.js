import React from 'react';
import useWindowSize from 'hooks/useWindowSize';
import Container from 'components/particles/Container';
import Button from 'components/atoms/Button';
import animations from './HeroContent.animations';
import S from './HeroContent.styles';

const HeroContent = () => {
  const size = useWindowSize();

  const buttonsAnimation = animations.getButtons(size);
  return (
    <Container axis="both">
      <S.Content>
        <S.Heading tag="h1" homePage animation={animations.heading}>
          Zaistniej w sieci
        </S.Heading>
        <S.Paragraph animation={animations.paragraph}>
          <span>Zaistniej w świecie&nbsp;</span>
          <span>nieograniczonych możliwości</span>
        </S.Paragraph>
      </S.Content>

      <S.Buttons {...buttonsAnimation}>
        <Button type="link" path="/wycena" priority="primary">
          Bezpłatna wycena
        </Button>
        <Button type="link" path="/projekty" priority="secondary">
          Zrealizowane projekty
        </Button>
      </S.Buttons>
    </Container>
  );
};

export default HeroContent;
