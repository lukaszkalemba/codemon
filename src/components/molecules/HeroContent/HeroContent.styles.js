import styled from 'styled-components';
import { motion } from 'framer-motion';
import HeadingComponent from 'components/atoms/Heading';
import ParagraphComponent from 'components/atoms/Paragraph';
import ContainerComponent from 'components/particles/Container';

const Content = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 50px;
  padding: 0 30px;

  ${({ theme }) => theme.media.laptop`
    flex-direction: column-reverse;
    padding: 0;
    margin-top: 90px;
    align-items: flex-end;
  `}

  ${({ theme }) => theme.media.desktop`
    margin-top: 150px;
    margin-left: 0;
  `}
`;

const Heading = styled(HeadingComponent)`
  ${({ theme }) => theme.media.laptop`
    margin: 0.20em 0;
  `}

  ${({ theme }) => theme.media.largeDesktop`
    font-size: ${theme.font.size.heading.xxl};
    margin: 0.15em 0;
  `}
`;

const Paragraph = styled(ParagraphComponent)`
  display: flex;
  flex-direction: column;

  span {
    margin: 1px 0;
  }

  ${({ theme }) => theme.media.tablet`
    font-size:  calc(${theme.font.size.body.m});
  `}

  ${({ theme }) => theme.media.laptop`
    flex-direction: row;
    margin-right: -130px;
  `}
`;

const Buttons = styled(motion.div)`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: 20px;
  padding: 0 30px;
  position: relative;
  z-index: ${({ theme }) => theme.zindex.level1};

  ${({ theme }) => theme.media.tablet`
    margin-top: 40px;
  `}

  ${({ theme }) => theme.media.laptop`
    flex-direction: row;
    padding: 0;
    justify-content: flex-end;
    margin-right: 130px;
  `}
`;

const Container = styled(ContainerComponent)`
  ${({ theme }) => theme.media.laptop`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding-right: 21vw;
  `}
`;

export default { Content, Heading, Paragraph, Buttons, Container };
