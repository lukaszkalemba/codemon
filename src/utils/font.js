export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
