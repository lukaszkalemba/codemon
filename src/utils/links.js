export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
