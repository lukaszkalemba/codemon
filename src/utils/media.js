export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
