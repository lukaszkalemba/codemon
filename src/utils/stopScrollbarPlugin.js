import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
