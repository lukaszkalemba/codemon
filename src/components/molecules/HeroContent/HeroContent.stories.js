import React from 'react';
import HeroContent from './HeroContent';

export const heroContent = () => <HeroContent />;

export default {
  component: HeroContent,
  title: 'Molecules|HeroContent',
};
