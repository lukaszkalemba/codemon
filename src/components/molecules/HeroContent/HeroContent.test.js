import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import HeroContent from './HeroContent';

describe('<HeroContent />', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = render(<HeroContent />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
