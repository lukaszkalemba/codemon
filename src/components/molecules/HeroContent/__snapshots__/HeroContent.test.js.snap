// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroContent /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-bdnylx ggAaCM"
>
  <div
    class="sc-gKAblj ixJViW"
  >
    <h1
      class="sc-hKFyIo LDwAY sc-iCoHVE jCGsBf"
      style="opacity: 0; transform: translateX(-80px) translateZ(0);"
    >
      Zaistniej w sieci
    </h1>
    <p
      class="sc-jSFkmK hYnFXU sc-fujyUd kUoHkZ"
    >
      <span>
        Zaistniej w świecie 
      </span>
      <span>
        nieograniczonych możliwości
      </span>
    </p>
  </div>
  <div
    class="sc-pNWxx cPvtmN"
  >
    <a
      class="sc-gtssRu jvhPFt sc-dlnjPT fblnUy"
      href="/wycena"
      priority="primary"
    >
      <span>
        Bezpłatna wycena
      </span>
    </a>
    <a
      class="sc-gtssRu jvhPFt sc-dlnjPT bxdVPr"
      href="/projekty"
      priority="secondary"
    >
      <span>
        Zrealizowane projekty
      </span>
    </a>
  </div>
</div>
`;
